src/one_time/core.clj

```diff
@@ -4,6 +4,15 @@
             [one-time.totp  :as totp]
             [one-time.hotp  :as hotp]))
 
+(defn ^:private ct=
+  "Constant-time comparison for numeric types, via bytewise string equality"
+  [x y]
+  (let [x' (.getBytes (str x))
+        y' (.getBytes (str y))]
+    (and
+     (= (count x') (count y'))
+     (->> (map bit-xor x' y') (reduce +) (= 0)))))
+
```

Comment:
Why not
``` clojure
(defn ^:private ct=
   "Constant-time comparison for numeric types, via bytewise string equality"
   (and (= (count x) (count y))
        (->> (map #(bit-xor (int %1) (int %2)) x y )
             (reduce +)
             (= 0))))
```
Advantages:
* works for `String` `[int]` `[char]` `[byte]`
* limits further heap allocations

Disadvantages:
* problems with character encoding?